src/vs/workbench/contrib/extensions/electron-sandbox/remoteExtensionsInit.ts

```diff
@@ -62,49 +64,22 @@ export class InstallRemoteExtensionsContribution implements IWorkbenchContributi
-		const alreadyInstalledInRemote = await this.extensionManagementServerService.remoteExtensionManagementServer.extensionManagementService.getInstalled(ExtensionType.User);
 		const alreadyInstalledLocally = await this.extensionManagementServerService.localExtensionManagementServer.extensionManagementService.getInstalled(ExtensionType.User);
+		const alreadyInstalledRemotely = await this.extensionManagementServerService.remoteExtensionManagementServer.extensionManagementService.getInstalled(ExtensionType.User);
+		const extensionsToInstall = settingValue
+			.filter(id =>
+				alreadyInstalledLocally.some(ext => areSameExtensions(ext.identifier, { id }))
+				&& !alreadyInstalledRemotely.some(ext => areSameExtensions(ext.identifier, { id }))
+			);
 
-		const prereleaseExtensionInfo: InstallExtensionInfo[] = [];
-		const extensionInfo: InstallExtensionInfo[] = [];
-		for (const id of settingValue) {
-			const alreadyInstalled = alreadyInstalledInRemote.some(e => areSameExtensions(e.identifier, { id }));
-			if (alreadyInstalled) {
-				this.logService.trace(`Default remote extension '${id}' is already installed`);
-				continue;
-			}
-
-			const installedLocally = alreadyInstalledLocally.find(e => areSameExtensions(e.identifier, { id }));
-			if (!installedLocally) {
-				this.logService.trace(`Default remote extension '${id}' is not installed locally`);
-				continue;
-			}
-
-			const extension = galleryExtensions.find(e => areSameExtensions(e.identifier, { id }));
-			if (!extension) {
-				this.logService.warn(`Default remote extension '${id}' is not found`);
-				continue;
-			}
-
-			const installPreReleaseVersion = installedLocally.isPreReleaseVersion;
-			this.logService.trace(`Default remote extension '${id}' queued for install (pre-release=${installPreReleaseVersion})`);
-			(installPreReleaseVersion ? prereleaseExtensionInfo : extensionInfo).push({
-				extension, options: { installPreReleaseVersion },
-			});
+		if (!extensionsToInstall.length) {
+			return;
 		}
 
-		// Install pre-release extensions first to avoid a situation where:
-		// An extension without a pre-release (A) is installed first and depends on an extension that has a pre-release version (B)
-		// If this happens, the extension A may result in the installation of the stable version of B
-		if (prereleaseExtensionInfo.length) {
-			await Promise.allSettled(prereleaseExtensionInfo.map(e => this.extensionManagementServerService.remoteExtensionManagementServer!.extensionManagementService.installFromGallery(e.extension, e.options)));
-		}
-		if (extensionInfo.length) {
-			await Promise.allSettled(extensionInfo.map(e => this.extensionManagementServerService.remoteExtensionManagementServer!.extensionManagementService.installFromGallery(e.extension, e.options)));
-		}
+		const galleryExtensions = await this.extensionsWorkbenchService.getExtensions(extensionsToInstall.map(id => ({ id })), CancellationToken.None);
```

Comment:
With above suggestions, you do not need to get gallery extensions. `this.extensionsWorkbenchService.installInServer` will take care of it.

Reply:
oh nice, thank you!